### PR TITLE
Compare EAP branch to latest release

### PR DESCRIPTION
**Pull request by asodja**
For better stability of sanityCheck.

### CODE REVIEW

File: testing/architecture-test/src/changes/accepted-changes/accepted-public-api-changes.json [hunk 1]
```diff
@@ -1,4 +1,42 @@
 {
     "acceptedApiChanges": [
+        {
```
Comment (ov7a) [inline testing/architecture-test/src/changes/accepted-changes/accepted-public-api-changes.json:3]: 🤔 why do we need this change?

File: build-logic/binary-compatibility/src/test/groovy/gradlebuild/binarycompatibility/PublicAPIRulesTest.groovy [hunk 2]
```diff
@@ -485,6 +485,7 @@ class PublicAPIRulesTest extends Specification {
             Map<String, ?> getUserData() {
                 [
                     currentVersion: '11.38',
+                    currentMasterVersion: '11.38',
```
Comment (ov7a) [inline build-logic/binary-compatibility/src/test/groovy/gradlebuild/binarycompatibility/PublicAPIRulesTest.groovy:488]: 🤔 Should it be tested with currentMasterVersion != currentVersion?

File: testing/architecture-test/src/changes/accepted-changes/accepted-public-api-changes.json (hunk 1 above)
Comment (asodja) [reply testing/architecture-test/src/changes/accepted-changes/accepted-public-api-changes.json:3]: These are accepted api changes on lastest master commit we are based on.

Although these are cleaned with master version bump to 8.12 2 days ago, see: https://github.com/gradle/gradle/commit/ab0ab5c8ad26b95c33ef65c86f234cce1998ce6d, but EAP branch is not yet rebased to latest master.

File: build-logic/binary-compatibility/src/test/groovy/gradlebuild/binarycompatibility/PublicAPIRulesTest.groovy (hunk 2 above)
Comment (asodja) [reply build-logic/binary-compatibility/src/test/groovy/gradlebuild/binarycompatibility/PublicAPIRulesTest.groovy]: We can also change that yes

Comment (asodja) [reply build-logic/binary-compatibility/src/test/groovy/gradlebuild/binarycompatibility/PublicAPIRulesTest.groovy]: Done

**Review by ov7a — APPROVED** [review]

**Pull request merged**